File: bilibili-backend/model/video.go
package model

// Video 视频表
type Video struct {
	BaseModel
	UserID          uint64 `gorm:"not null;index:idx_user_created" json:"user_id"`
	Title           string `gorm:"type:varchar(200);not null" json:"title"`
	Description     string `gorm:"type:text" json:"description"`
	CoverURL        string `gorm:"type:varchar(255)" json:"cover_url"`
	VideoURL        string `gorm:"type:varchar(255)" json:"video_url"`
	TranscodedURL   string `gorm:"type:varchar(255)" json:"transcoded_url"`
	Category        string `gorm:"type:varchar(50);index:idx_cat_status_created,priority:1" json:"category"`
	Status          int    `gorm:"type:tinyint;default:2;index:idx_cat_status_created,priority:2" json:"status"` // 1=已发布 2=转码中
	TranscodeStatus int    `gorm:"type:tinyint;default:0" json:"transcode_status"`                               // 0=未开始 1=转码中 2=转码完成 3=转码失败
	ViewCount       int64  `gorm:"default:0" json:"view_count"`
	LikeCount       int    `gorm:"default:0" json:"like_count"`
	CommentCount    int    `gorm:"default:0" json:"comment_count"`
	DanmakuCount    int    `gorm:"default:0" json:"danmaku_count"`
}

File: bilibili-frontend/src/types/video.ts
// =========================
// 视频模块 TypeScript 类型定义
// =========================

/** 视频上传表单 */
export interface UploadForm {
  file: File
  title: string
  description?: string
  category: string
  cover?: File
}

/** 视频上传响应 */
export interface UploadResponse {
  id: number
  title: string
  cover_url: string
  status: number
  transcode_status: number
  created_at: string
}

/** 转码状态响应 */
export interface TranscodeStatusResponse {
  transcode_status: number
  transcoded_url: string
}

/** 视频列表项 */
export interface VideoListItem {
  id: number
  title: string
  cover_url: string
  category: string
  view_count: number
  like_count: number
  created_at: string
  user?: {
    id: number
    username: string
    nickname: string
    avatar: string
  }
}

/** 视频列表响应 */
export interface VideoListResponse {
  list: VideoListItem[]
  total: number
  page: number
  size: number
}

/** 视频详情 */
export interface VideoDetail {
  id: number
  title: string
  description: string
  cover_url: string
  video_url: string
  transcoded_url: string
  category: string
  status: number
  transcode_status: number
  view_count: number
  like_count: number
  comment_count: number
  danmaku_count: number
  created_at: string
  user_info: {
    id: number
    username: string
    nickname: string
    avatar: string
  }
}

/** 分类选项 */
export interface CategoryOption {
  id: string
  label: string
  code: string
}

export const videoCategories: CategoryOption[] = [
  { id: 'cine', label: '影视', code: '[CINE]' },
  { id: 'game', label: '游戏', code: '[GAME]' },
  { id: 'acgn', label: '二次元', code: '[ACGN]' },
  { id: 'chef', label: '美食', code: '[CHEF]' },
  { id: 'docu', label: '纪录片', code: '[DOCU]' },
  { id: 'show', label: '综艺', code: '[SHOW]' },
]
